index.js

```diff
@@ -75,13 +78,19 @@ export default class QRCodeScanner extends Component {
     this._handleBarCodeRead = this._handleBarCodeRead.bind(this);
   }
 
-  componentWillMount() {
+  async componentWillMount() {
     if (Platform.OS === 'ios') {
       Camera.checkVideoAuthorizationStatus().then(isAuthorized => {
         this.setState({ isAuthorized })
       })
-    }
-    else {
+    } else if (Platform.OS === 'android') {
+      const granted = await PermissionsAndroid.request( PermissionsAndroid.PERMISSIONS.CAMERA, { 
+          'title': this.props.permissionDialogTitle, 
+          'message':  this.props.permissionDialogMessage, 
+        }
+      ) 
+      this.setState({ isAuthorized: granted ===  PermissionsAndroid.RESULTS.GRANTED })
+    } else {
```

Comment:
can we do this in a promise instead, the rationale here being that many clients using this library may not have async/await configured, and since we don't have a build step to produce a transpiled version, `async/await` may not be supported and could be a breaking change.